### PR TITLE
LP-29 osgearth integration (utitilies + osgearth lib)

### PR DESCRIPTION
**Pull request by filnet**
Please review the changes in :
- libs/utils (first two commits).
- everything but libs/osgearth/* in the last commit

You can also review libs/osgearth/* if you feel like it but it is a lot of new code

### CODE REVIEW

File: ground/openpilotgcs/src/libs/libs.pro [hunk 1]
```diff
@@ -13,4 +13,6 @@ SUBDIRS   = \
     qwt \
     sdlgamepad
 
-SUBDIRS +=
+exists( $(OSG_SDK_DIR) ) {
```
Comment (parched) [inline ground/openpilotgcs/src/libs/libs.pro]: This isn't going to work via Qt creator.

Comment (filnet) [inline ground/openpilotgcs/src/libs/libs.pro]: Yes, you mentioned that before.
What would work with QtCreator?
PS : I use QtCreator only to edit ui files.

Comment (parched) [inline ground/openpilotgcs/src/libs/libs.pro]: because that environment variable won't be defined because it doesn't use top level makefile

Comment (filnet) [inline ground/openpilotgcs/src/libs/libs.pro]: is it fixable at my end?

**Pull request merged**